Pranav/timer

pratri opened the pull request.

pratri committed:
testing error

pratri committed:
Added Timer counting up and formatted it

pratri committed:
I guess I missed this one

taliyos committed:
Removed old timer.h/timer.cpp from Cmake

taliyos committed:
Simplify timer logic

taliyos committed:
Removed extra time

pratri committed:
Merge pull request #7 from taliyos/charles/timer-tweaks

Simplify timer logic

pratri committed:
Pause Resume Button functionality

Now the button to the right of the timer will correctly resume and pause the timer.

pratri committed:
Added split functionality and keyboard input

Added split function in split.cpp + splitlistdata.cpp which sends a signal to spitmodel.cpp which updates the ui by calling dataChanged.
Added keyboard inputs to pause play and split functions.
Added forceActiveFocus to return focus to main app(so the buttons work) after finishing editing split.

pratri committed:
Hardcoded Hotkeys for Pause, Split , and Reset

taliyos commented on Components/Fonts/OpenSans.qml:
```diff
@@ -1,4 +1,4 @@
-pragma Singleton
+        pragma Singleton
```
Please revert this file

taliyos commented on Components/Split/split.cpp:
```diff
@@ -73,7 +100,7 @@ QString Split::getGameName() const {
 void Split::setGameName(const QString& name) {
     m_layout->gameName = name;
-    emit gameNameChanged();
+    // emit gameNameChanged();
```
Why is this commented?

taliyos commented on Components/Split/split.h:
```diff
@@ -43,6 +46,10 @@ class Split : public QObject {
     void setAttemptCount(const int& attemptCount);
 
+
+    // void setTimer();
```
Please remove if not needed

taliyos commented on Components/Split/split.h:
```diff
@@ -56,10 +63,16 @@ class Split : public QObject {
     void openFile(const QString& fileLocation);
     void newFile();
+    Timer* getTimer();
+    void onResetButtonPress();
```
Move getTimer() below other void funtions

taliyos commented on Components/Split/split.h:
```diff
@@ -56,10 +63,16 @@ class Split : public QObject {
     SplitListData* m_data;
+    Timer* m_timer;
+
+    int splitrow = 0;
```
Change splitrow -> m_splitRow to follow the naming structure

taliyos commented on Components/SplitList/splitlistdata.h:
```diff
@@ -26,15 +26,20 @@ class SplitListData : public QObject {
     void removeItem(int index);
     void clear();
 
+    void setTimeatSplitIndex(const QString& time, int index);
```
Change name to setTimeAtSplitIndex

taliyos commented on Components/Timer/hotkeymanager.cpp:
```diff
@@ -0,0 +1,26 @@
+#include "hotkeymanager.h"
```
Move hotkey manager outside of timer component. It's operating globally and can also affect splits.

taliyos commented on Components/Timer/timer.cpp:
```diff
@@ -0,0 +1,100 @@
+#include <unistd.h>
+#include <qDebug>
+
+//
```
Remove empty comment

taliyos commented on Components/Timer/timer.cpp:
```diff
@@ -0,0 +1,100 @@
+    }else{
+        elapsedTimer = pausedTime + (elapsedTimer - pausedTime) - deadTime;
+    }
+    qint64 elapsedTimerRounded = (elapsedTimer)/ 10;
```
Suggestion: Move these conversion steps to a separate function

taliyos commented on Components/Timer/timer.cpp:
```diff
@@ -0,0 +1,100 @@
+    pausedTime = 0;
+    resumedTime = 0;
+    deadTime = 0;
+    // timer.invalidate();
```
Is the commented code needed?

taliyos commented on Components/Timer/timer.h:
```diff
@@ -0,0 +1,45 @@
+public:
+    // To attempt to fix type_traits: to help -> about plugins -> deactivate clang code model
+    explicit Timer(const int& updateMilisecondInterval, QObject* parent = nullptr);
+    // ~Timer() override;
```
Please cleanup commented functions

taliyos replied:
Additionally, move the parent constructor parameter in front of the int.

pratri replied:
What does move parent constructor parameter in front of int mean?
Is it this line:
explicit Timer(const int& updateMilisecondInterval, QObject* parent = nullptr);
To be this:
explicit Timer(QObject* parent = nullptr, const int& updateMilisecondInterval);

If so that errors

taliyos commented on Components/Timer/timer.h:
```diff
@@ -0,0 +1,45 @@
+
+    static QString formatTime(QList<qint64> timeArray, int index, const QString& separator);
+
+    // True when timer paused, false when unpaused
```
This comment seems unnecessary

taliyos reviewed (changes requested):
Some code cleanup is needed. Please take a look at the comments and make the appropriate changes. Also, please re-add the font. I'm not sure why that was removed.

Also, make sure to pull in the latest and resolve the merge conflicts.

pratri committed:
Code Cleanup and fixes

pratri committed:
Resources.qrc fix

pratri committed:
Merge branch 'main' into pranav/timer

taliyos committed:
Fix fonts

taliyos committed:
Remove/simplify theming

pratri committed:
Merge pull request #10 from taliyos/charles/style-fix

Remove/simplify theming

pratri committed:
Fixes

Timer format fixed
Added Finish
Added green color when finishing a run
fixed a bug with split index if no splits existed

pratri committed:
Added Server Functionality
